src/Entities/Project.php

```diff
@@ -11,14 +13,32 @@ class Project
     private int $client_id;
     private ?DateTime $deadline;
     private ?bool $isOverdue;
+    private ?string $client_name;
+    private ?string $client_logo;
+    private ?array $users;
 
-    public function __construct(int $id, string $name, int $client_id, ?string $deadline)
+    public function __construct(int $id, string $name, int $client_id, ?string $deadline, ?string $client_name = null, ?string $client_logo = null, ?array $users = null)
     {
+        if ($users != null) {
+            $this->validateUsersArray($users);
+        }
         $this->id = $id;
         $this->name = $name;
         $this->client_id = $client_id;
-        $this->deadline =  is_null($deadline) ? null : new DateTime($deadline);
+        $this->deadline = is_null($deadline) ? null : new DateTime($deadline) ;
         $this->isOverdue = $this->calculateIsOverdue();
+        $this->client_name = $client_name;
+        $this->client_logo = $client_logo;
+        $this->users = $users;
+    }
+
+    private function validateUsersArray(array $users)
+    {
+        foreach ($users as $user) {
+            if (!($user instanceof User)) {
+                throw new InvalidUserArrayDatatypeException("Invalid user array\n");
```

Comment:
this error message could be more informative, "Invalid user array" doesn't necessarily explain what happened very clearly

Reply:
done